status-handler.js

```diff
@@ -161,10 +167,7 @@ class StatusHandler {
    * @returns {Promise} resolves to boolean whether we are complete or not
    */
   async isComplete(spec) {
-    const { StatusCounter, Status } = this.models;
-    const [{ count }, { total }] = await Promise.all([StatusCounter, Status].map(m => m.findOne(spec)));
-
-    const progress = (count / total) * 100;
+    const progress = await this.progress.compute(spec);
     return progress === 100;
```

Comment:
that exact equals bothers me, given it's backed by maths.  I know it's just a simple divide, but...  maybe (per my other comment) the `progress.compute` fn should return the triplet of data to make this less... squishy.

Reply:
I'd also do a `>=` just in case...